File: c_language/easy/509.fibonacci.c
int fib(int n){
  if (n == 0) return n;
  int n1 = 0, n2 = 1, temp;

  for (int i = 0; i < n-1; i++) {
    temp = n2;
    n2 += n1;
    n1 = temp;
  }
  return n2;
}

File: 1.easy/509.fibonacci.py
class Solution:
    def fib(self, n: int):
        n1 = 1
        n2 = 1
        while n > 2:
            n -= 1
            n2 += n1
            n1 = n2 - n1
        return 0 if (n == 0) else n2

n = int(input())
print(Solution.fib(None, n))